src/garage/sampler/config.py

```diff
@@ -0,0 +1,217 @@
+            env_infos[k].append(v)
+        path_length += 1
+        if d:
+            break
+        o = next_o
+    for k, v in agent_infos.items():
+        agent_infos[k] = np.asarray(v)
+    for k, v in env_infos.items():
+        env_infos[k] = np.asarray(v)
+    return (worker_number, np.array(observations), np.array(actions),
+            np.array(rewards), dict(agent_infos), dict(env_infos))
+
+
+def default_env_update_fn(config, worker_number, env, env_update):
+    """Use any non-None env_update as a new environment.
+
+    A simple env update function. If env_update is not None, it should be the
+    complete new environment. Otherwise, the old environment is returned.
+
+    This allows changing environments by passing the new environment as
+    `env_update` into `obtain_samples`.
+
+    Args:
+        config(SamplerConfig): The config this function is stored in. This
+            argument is ignored, but present to match the API set by
+            `SamplerConfig`.
+        worker_number(int): The number of the worker this update is occurring
+            in. This argument is ignored, but present to match the API set by
+            `SamplerConfig`.
+        env(gym.Env): The environment to update.
+        env_update(gym.Env or None): The environment to replace it with.
+
+    Returns:
+        The updated environment.
+
+    """
+    if env_update is not None:
+        return env_update
+    else:
+        return env
+
+
+def default_agent_update_fn(config, worker_number, agent, agent_update):
+    """Update an agent, assuming it implements garage.tf.policies.Policy2.
+
+    Args:
+        config(SamplerConfig): The config this function is stored in. This
+            argument is ignored, but present to match the API set by
+            `SamplerConfig`.
+        worker_number(int): The number of the worker this update is occurring
+            in. This argument is ignored, but present to match the API set by
+            `SamplerConfig`.
+        agent(garage.tf.policies.Policy2): The agent to update.
+        agent_update(dict[str, np.array]): Parameters to agent, which should
+            have been generated by calling `policy.get_param_values`.
+
+    Returns:
+        The updated agent.
+
+    """
+    agent.set_param_values(agent_update)
+    return agent
+
+
+def default_worker_init_fn(config, worker_number):
+    """Initialize a worker.
+
+    Args:
+        worker_number(int): The number of the worker this update is occurring
+            in. This argument is used to set a different seed for each worker.
+        config(SamplerConfig): The config this function is stored in. Used to
+            get the main seed, on which all other seeds used by the sampler are
+            based.
+
+    """
+    deterministic.set_seed(config.seed + worker_number)
+
+
+def identity_function(x):
+    """Do nothing.
+
+    This function can exists so it can be pickled.
+    """
+    return x
+
+
+class SamplerConfig:
```

Comment:
SamplingStrategy? SamplingPlan?

Reply:
WorkerFactory? The name is still up for discussion, of course.